src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_cthun.cpp

```diff
@@ -585,6 +599,43 @@ struct boss_cthun : public BossAI
 
         //Body Phase
         uint8 _fleshTentaclesKilled;
+
+        //If no-one is outside
+        uint32 _acidTimer;
+
+        bool IsEveryoneAffectedByDigestiveAcid() const
+        {
+            Map* map = me->GetMap();
```

Comment:
Rather than do this bit of logic every time, can you create a private Map* variable that is initiated on AI creation that you reference?

Reply:
```cpp
me->GetMap()->DoForAllPlayers([&](Player* player)
{
    ....
}

Reply:
For some reason conditioning using this didn't work, however casting spell did